blocks/gu_spoverview/tests/spoverview_test.php

```diff
@@ -0,0 +1,251 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Events tests.
+ *
+ * @package    block_spoverview
+ * @category   test
+ * @copyright  2020 Alejandro De Guzman <a.g.de.guzman@accenture.com>
+ */
+
+defined('MOODLE_INTERNAL') || die();
+
+global $CFG;
+
+require_once('config.php');
+require_once($CFG->dirroot .'/blocks/moodleblock.class.php');
+require_once($CFG->dirroot .'/blocks/gu_spoverview/block_gu_spoverview.php');
+
+
+class block_gu_spoverview_testcase extends advanced_testcase {
+
+    /**
+     * Setup test data.
+     */
+    public function setUp(){
+        global $DB;
+
+        $this->spoverview = new block_gu_spoverview();
+    
+        $this->resetAfterTest(true);
+
+        $gen = $this->getDataGenerator();
+
+        $this->student = $gen->create_user();
+        $this->teacher = $gen->create_user();
+
+        $this->category = $gen->create_category();
+
+        $this->course = $gen->create_course(array(
+                                    'name'     => 'Sample course', 
+                                    'category' => $this->category->id
+                                )
+                        );
+        $gen->enrol_user($this->student->id, $this->course->id, 'student');
+        $gen->enrol_user($this->teacher->id, $this->course->id, 'editingteacher');
+
+        $gen->create_module('assign', array(
+            'course' => $this->course->id,
+            'duedate' => time() * 30 * 30,
+            'status' => 'new'
+        ));
+
+        $gen->create_module('assign', array(
+            'course' => $this->course->id,
+            'duedate' => time() - 10,
+            'status' => 'new'
```

Comment:
'assign' module does not have status column. Status is found in assign_submission. Please create an insert_record('assign_submission') instead.